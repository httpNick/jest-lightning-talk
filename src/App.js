import React, { Component } from 'react';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {elementList : []};
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(
      {
        elementList : this.state.elementList.concat(["element " + this.state.elementList.length])
      }
    );
  }

  render() {
    return (
      <div className="App">
        <button onClick={this.handleClick}>Click me</button>
        <ul>
          {this.state.elementList.map(x => <li key={x}>{x}</li>)}
        </ul>
      </div>
    );
  }
}

export default App;
