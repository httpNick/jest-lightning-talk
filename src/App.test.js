import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('adds element when button is clicked', () => {

  const component = renderer
    .create(<App />);

  let tree = component
    .toJSON();

  let button = tree.children
    .find(x => x.type === 'button');

  console.log(button);

  expect(tree)
    .toMatchSnapshot();

  button.props
    .onClick();

  tree = component
    .toJSON();
  expect(tree)
    .toMatchSnapshot();

});
