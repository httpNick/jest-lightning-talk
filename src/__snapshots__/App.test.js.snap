exports[`test adds element when button is clicked 1`] = `
<div
  className="App">
  <button
    onClick={[Function bound handleClick]}>
    Click me
  </button>
  <ul />
</div>
`;

exports[`test adds element when button is clicked 2`] = `
<div
  className="App">
  <button
    onClick={[Function bound handleClick]}>
    Click me
  </button>
  <ul>
    <li>
      element 0
    </li>
  </ul>
</div>
`;
